proposed/2023/sponsor-link.md

```diff
@@ -0,0 +1,164 @@
+# Sponsor links for NuGet packages
+<!-- Replace `Title` with an appropriate title for your design -->
+
+- Jon Douglas <!-- GitHub username link -->
+- [#9339](https://github.com/NuGet/NuGetGallery/issues/9339) <!-- GitHub Issue link -->
+
+## Summary
+
+<!-- One-paragraph description of the proposal. -->
+
+Today, more than 9/10 companies use open source software in at least some capacity. One key concern is ensuring that open source is financially sustainable. More specifically, how do we make sure the developers maintaining some of the most critical software being used by these companies, governments, nonprofits, etc are able to keep doing their important work?
+
+This proposal introduces an end-to-end experience of bringing visibility to the sponsorship requirements of open source packages on the NuGet.org registry and make NuGet tooling aware of it. This is to ensure that the sponsorship requirements make it down to the developers who are using open source software in a non-disruptive way.
+
+Package authors will be able to express their sponsorship needs through new `sponsor` package metadata specifying a popular verified funding mechanism (GitHub Sponsors, Patreon, etc). This may consist of a list of reputable funding providers such as patreon, tidelift, github sponsors, and the relative link to each.
+
+Package consumers will be able to see packages looking for funding through different mechanisms. One will be at `restore` time where a new message will be outputted to summarize all the funding requirements in the individual has opt-in to a feature flag such as `<NuGetSponsor>true</NuGetSponsor>`. For more information about any specific package's sponsorship needs, a developer will be able to use a CLI command like `dotnet sponsor package <PACKAGE_NAME>` to learn more about how they can contribute financially. Within an IDE such as Visual Studio, there will be a new sponsorship affordance in the package details page that a developer can click on.
```

Comment:
Can `<NuGetSponsor>true</NuGetSponsor>` be included in the default `csproj` template? Similar to how `ImplicitUsings` and `Nullable` were introduced (on by default for new projects, but opt-in in principle).

Reply:
Would also be nice to have something similar to `npm fund`. It can generate a tree of your project dependencies with sponsor links for each organization/author.

![image](https://github.com/NuGet/Home/assets/1935960/41bf3159-825a-46b7-9d25-7d227cbcda91)

Reply:
> Can `<NuGetSponsor>true</NuGetSponsor>` be included in the default `csproj` template? Similar to how `ImplicitUsings` and `Nullable` were introduced (on by default for new projects, but opt-in in principle).

See the future possibilities section.

> Would also be nice to have something similar to `npm fund`. It can generate a tree of your project dependencies with sponsor links for each organization/author.
> 
> ![image](https://user-images.githubusercontent.com/1935960/272972106-41bf3159-825a-46b7-9d25-7d227cbcda91.png)

See the section showing the list command. Ideally you would use --include-transitives as well to show a similar graph.